resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesConf.scala

```diff
@@ -78,7 +78,10 @@ private[spark] class KubernetesDriverConf(
 
   override val resourceNamePrefix: String = {
     val custom = if (Utils.isTesting) get(KUBERNETES_DRIVER_POD_NAME_PREFIX) else None
-    custom.getOrElse(KubernetesConf.getResourceNamePrefix(appName))
+    val resourceNamePrefix = custom.getOrElse(KubernetesConf.getResourceNamePrefix(appName))
+    // If the first character of resourceNamePrefix is number,add the extra prefix : "spark-".
+    val prefix = "spark-" + resourceNamePrefix.charAt(0)
+    resourceNamePrefix.replaceAll("^[0-9]", prefix)
```

Comment:
Why not just prepend "spark-" if the first character is a digit? `if (Character.isDigit(resourceNamePrefix(0)))`

Reply:
> Why not just prepend "spark-" if the first character is a digit? `if (Character.isDigit(resourceNamePrefix(0)))` But don't you actually need to check if it starts with a-z?




------Hi, this code does not change the resourceNamePrefix that the first character is not a digit.


For example: resourceNamePrefix = 1min-xxx
After execute this code:

val prefix = "spark-" + resourceNamePrefix.charAt(0) resourceNamePrefix.replaceAll("^[0-9]", prefix)

**resourceNamePrefix = spark-1min-xxx,**
which satisfies the regular expression `[a-z]([-a-z0-9]*[a-z0-9])?`

For example : resourceNamePrefix = min-xxx
After execute this code:

val prefix = "spark-" + resourceNamePrefix.charAt(0) resourceNamePrefix.replaceAll("^[0-9]", prefix)

**resourceNamePrefix = min-xxx,which does not change.**

Local testing was done for thisoverride val resourceNamePrefix only.

Reply:
I know, it's just more complex than it needs to be, and doesn't catch all the cases that would fail the pattern you show above: `[a-z]([-a-z0-9]*[a-z0-9])?)`